crates/sui-core/src/epoch/randomness.rs

```diff
@@ -0,0 +1,512 @@
+            fastcrypto_tbls::ecies::PrivateKey::<bls12381::G2Element>::from(randomness_private_key),
+            nodes,
+            t.into(),
+            fastcrypto_tbls::random_oracle::RandomOracle::new(prefix_str.as_str()),
+            &mut rand::thread_rng(),
+        ) {
+            Ok(party) => party,
+            Err(err) => {
+                error!("random beacon: error while initializing Party: {err:?}");
+                return None;
+            }
+        };
+        info!(
+            "random beacon: state initialized with authority={name}, total_weight={total_weight}, t={t}, num_nodes={num_nodes}, oracle initial_prefix={prefix_str:?}",
+        );
+
+        // Load existing data from store.
+        let manager = Self {
+            epoch_store: epoch_store_weak,
+            consensus_adapter,
+            party,
+            processed_messages: Mutex::new(BTreeMap::new()),
+            used_messages: OnceCell::new(),
+            confirmations: Mutex::new(BTreeMap::new()),
+            dkg_output: OnceCell::new(),
+        };
+        let dkg_output = tables
+            .dkg_output
+            .get(&SINGLETON_KEY)
+            .expect("typed_store should not fail");
+        if let Some(dkg_output) = dkg_output {
+            info!(
+                "random beacon: loaded existing DKG output for epoch {}",
+                committee.epoch()
+            );
+            epoch_store
+                .metrics
+                .epoch_random_beacon_dkg_num_shares
+                .set(dkg_output.shares.as_ref().map_or(0, |shares| shares.len()) as i64);
+            manager
+                .dkg_output
+                .set(dkg_output)
+                .expect("setting new OnceCell should succeed");
+        } else {
+            info!(
+                "random beacon: no existing DKG output found for epoch {}",
+                committee.epoch()
+            );
+            // Load intermediate data.
+            manager.processed_messages.lock().unwrap().extend(
+                tables
+                    .dkg_processed_messages
+                    .safe_iter()
+                    .map(|result| result.expect("typed_store should not fail")),
+            );
+            if let Some(used_messages) = tables
+                .dkg_used_messages
+                .get(&SINGLETON_KEY)
+                .expect("typed_store should not fail")
+            {
+                manager
+                    .used_messages
+                    .set(used_messages.clone())
+                    .expect("setting new OnceCell should succeed");
+            }
+            manager.confirmations.lock().unwrap().extend(
+                tables
+                    .dkg_confirmations
+                    .safe_iter()
+                    .map(|result| result.expect("typed_store should not fail")),
+            );
+        }
+
+        Some(manager)
+    }
+
+    pub fn start_dkg(&self) -> SuiResult<()> {
```

Comment:
you don't need to change this but for future reference, SuiResult has defaults for both params so you don't need `<()>`

Reply:
good to know! cleaned it up